### ISSUE TITLE
Screen unblank on physical button press?

### ISSUE THREAD

**Issue by kchiem** [Blackymas/NSPanel_HA_Blueprint#1281]
Should the screen exit from the screensaver when one of the physical buttons is pressed?

**Comment by kchiem** [comment]
Can there be a setting to wake up on short presses?

**Comment by edwardtfn** [comment]
I don't have the panel with me for the next few days, but please try adding this to your yaml and install it in your panel:

```yaml
binary_sensor:
  - id: !extend left_button
    on_click:
      - lambda: |-
          disp1->goto_page("home");
  - id: !extend right_button
    on_click:
      - lambda: |-
          disp1->goto_page("home");
```

Please let me know the results.

**Comment by kchiem** [comment]
It works!  Although if the screen is already on, pressing a button causes a slight blinking effect with the screen.  If that could be avoided, that'd be great.

**Comment by edwardtfn** [comment]
Try this:

```yaml
binary_sensor:
  - id: !extend left_button
    on_click:
      - lambda: |-
          if (current_page->state == "screensaver") disp1->goto_page("home");
  - id: !extend right_button
    on_click:
      - lambda: |-
          if (current_page->state == "screensaver") disp1->goto_page("home");
```

**Comment by kchiem** [comment]
This fixes the blinking, thanks!  Any plans on making this default?

**Comment by edwardtfn** [comment]
> Any plans on making this default?

No. 😂


I'm sure this would be controversial...
Why do you want this as default? These customizations usually supports new versions pretty well without need from changes from your side, so I would suggest to add in the [customization docs](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Customization) instead, or even add to some build tests, so we know when it is broken after some change... 😉

**Comment by kchiem** [comment]
> Why do you want this as default?

It seems logical to me that if you use the panel in some way, the screen should turn on.  It already does that when the screen is touched, why not when the buttons are pressed?

Not sure I understand what you mean by the customization docs or the build tests.

Can the Timeout Sleep range be expanded?  It's limited to 300s right now.  I have these panels in bathrooms, and most of the time, they're only occupied for 10-20 mins, so I'd like to be able to keep the screen on for that period.  Otherwise, perhaps a feature like keeping the screen on if either relays are on?

**Comment by edwardtfn** [comment]
> Can the Timeout Sleep range be expanded? It's limited to 300s right now. I have these panels in bathrooms, and most of the time, they're only occupied for 10-20 mins, so I'd like to be able to keep the screen on for that period. Otherwise, perhaps a feature like keeping the screen on if either relays are on?

Try adding this to your yaml:
```yaml
number:
  - id: !expand timeout_sleep
    max_value: 1500
```

**Comment by kchiem** [comment]
```
could not determine a constructor for the tag '!expand'
  in "/config/esphome/nspanel-br3.yaml", line 44, column 9:
      - id: !expand timeout_sleep
            ^
```

**Comment by edwardtfn** [comment]
My fault... Please try this:
```yaml
number:
  - id: !extend timeout_sleep
    max_value: 1500
```

**Comment by kchiem** [comment]
This works.  Thanks again!

**Comment by Xitro01** [comment]
> > Why do you want this as default?
> 
> It seems logical to me that if you use the panel in some way, the screen should turn on. It already does that when the screen is touched, why not when the buttons are pressed?
> 
> Not sure I understand what you mean by the customization docs or the build tests.
> 
> Can the Timeout Sleep range be expanded? It's limited to 300s right now. I have these panels in bathrooms, and most of the time, they're only occupied for 10-20 mins, so I'd like to be able to keep the screen on for that period. Otherwise, perhaps a feature like keeping the screen on if either relays are on?

Is it a smart bulb in the bathroom? I hacked something that if the room light is on the timeout = 0 and if the light is off the timeout = 5. That combined with the if a button is pressed the screened should go on, what [edwardtfn](https://github.com/edwardtfn) mentioned.

**Comment by edwardtfn** [comment]
> @edwardtfn is there a way to delay turning on the display for 10ms after clicking? Sometimes the nspanel doesn't recognize the light is on straight away (so the bar stays white instead of light blue).

Looks like you found a bug. Which version are you running?

**Comment by Xitro01** [comment]
> > @edwardtfn is there a way to delay turning on the display for 10ms after clicking? Sometimes the nspanel doesn't recognize the light is on straight away (so the bar stays white instead of light blue).
> 
> Looks like you found a bug. Which version are you running?

Version 4.1.4
I flashed the device 2 days ago and this is my first nspanel, so I've got everything to the (current) latest version.

**Comment by edwardtfn** [comment]
Ok. I will try to duplicate this here.

**Comment by edwardtfn** [comment]
I believe that fix to v4.2 will solve this issue.

**Comment by edwardtfn** [comment]
- #1556
